plugins/st2/auth_token.go

```diff
@@ -0,0 +1,66 @@
+package st2
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func AuthToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.AuthToken,
+		DocsURL:       sdk.URL("https://docs.stackstorm.com/authentication.html"),
+		ManagementURL: sdk.URL("https://docs.stackstorm.com/authentication.html"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to StackStorm.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 32,
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
```

Comment:
<img width="1341" alt="Screenshot 2023-03-30 at 10 26 29" src="https://user-images.githubusercontent.com/45151601/228776279-62ace7c9-6f44-45e2-94ba-5c1d3ffbecb1.png">

I found this in the documentation of `st2`. We should take this into account, as, if I understand it correctly, it otherwise it would basically cache the first token it receives. 

Is this something that would have to be done manually by the user, as a step of setting up the plugin?

Reply:
Link: https://docs.stackstorm.com/reference/cli.html

Reply:
For context: Shell Plugins aim to get rid of the need to store secrets in plain-text on disk. My concern is that with the cache enabled, this would not be the case.

Reply:
> Is this something that would have to be done manually by the user, as a step of setting up the plugin?

To the best of my knowledge, setting this would be up to the user.

Reply:
> For context: Shell Plugins aim to get rid of the need to store secrets in plain-text on disk. My concern is that with the cache enabled, this would not be the case.

Yes, without `cache_token: False`, after logging in, `~/.st2/token-<username>` gets created and looks like this:

```shell
❯ cat ~/.st2/token-mark
{"token": "thisisthetokenhunter2", "expire_timestamp": 1680391552}
```

Reply:
Perhaps relying on the config file `~/.st2/config` is the best path forward?

```
[credentials]
# Credentials used to authenticate against the auth API and retrieve the auth
# token
username = test1
password = testpassword
```

Reply:
I've continued this thread [here](https://github.com/1Password/shell-plugins/pull/172#issuecomment-1523207921).